### PR TITLE
Construct the test runtime against filtered OSGi/PDE metadata

### PR DESCRIPTION
**Pull request by lppedd**
This completes what has been done in #1625.

@laeubi let me know if you think this might arise some sort if issues. 
Per my testing, it's perfectly backward compatible.

Draft for now as I'm open to discussion.

### CODE REVIEW

File: tycho-surefire/tycho-surefire-plugin/src/main/java/org/eclipse/tycho/surefire/AbstractTestMojo.java [hunk 1]
```diff
@@ -699,6 +701,9 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             return;
         }
         if (shouldRun()) {
+            // Allow constructing the test runtime against filtered OSGi/PDE metadata
+            project.setContextValue(TychoConstants.CTX_METADATA_ARTIFACT_LOCATION, metadataDirectory);
```
Comment (laeubi) [inline tycho-surefire/tycho-surefire-plugin/src/main/java/org/eclipse/tycho/surefire/AbstractTestMojo.java:705]: One should not use the project to store values, please always use ReactorProject for this

Comment (lppedd) [reply tycho-surefire/tycho-surefire-plugin/src/main/java/org/eclipse/tycho/surefire/AbstractTestMojo.java:705]: @laeubi that's fine! I will have to go back and change the `MavenProject` usage in the other PR too then.

Comment (laeubi) [reply tycho-surefire/tycho-surefire-plugin/src/main/java/org/eclipse/tycho/surefire/AbstractTestMojo.java:705]: You can jsut include all changes in this PR

Comment (lppedd) [reply tycho-surefire/tycho-surefire-plugin/src/main/java/org/eclipse/tycho/surefire/AbstractTestMojo.java]: @laeubi done! Should be ok now.

**Review by laeubi — APPROVED** [review]
LGTM

**Pull request merged**